src/odemis/gui/cont/acquisition.py

```diff
@@ -838,15 +841,16 @@ def _on_zstack(self):
         """
         if not self._zStackActive.value:
             return
-        self._zlevels = {}
-        for s in self._acquiStreams.value:
-            # the other possible streams are SEM which are generally not wanted by the user to be zstacked  
-            if isinstance(s, (FluoStream, BrightfieldStream)):
-                levels = generate_zlevels(
-                    self._tab_data.main.focus,
-                    [self._tab_data.zMin.value, self._tab_data.zMax.value],
-                    self._tab_data.zStep.value)
-                self._zlevels[s] = levels
+
+        levels = generate_zlevels(
+            self._tab_data.main.focus,
+            [self._tab_data.zMin.value, self._tab_data.zMax.value],
+            self._tab_data.zStep.value)
```

Comment:
I think adjusting the tabs like this would make it more readable:
```
levels = generate_zlevels(
                          self._tab_data.main.focus,)
                          [self._tab_data.zMin.value, self._tab_data.zMax.value],
                          self._tab_data.zStep.value)
```